lib/mongoose_push_web/plugs/cast_and_validate.ex

```diff
@@ -0,0 +1,133 @@
+defmodule MongoosePushWeb.Plug.CastAndValidate do
+  @behaviour Plug
+
+  @impl Plug
+  def init(opts) do
+    OpenApiSpex.Plug.CastAndValidate.init(opts)
+  end
+
+  @impl Plug
+  def call(conn = %{params: %{"service" => _, "alert" => _}}, opts = %{operation_id: _}) do
+    OpenApiSpex.Plug.CastAndValidate.call(conn, opts)
+  end
+
+  @impl Plug
+  def call(conn = %{params: %{"service" => _, "data" => _}}, opts = %{operation_id: _}) do
+    OpenApiSpex.Plug.CastAndValidate.call(conn, opts)
+  end
+
+  @impl Plug
+  def call(conn = %{params: %{"service" => _}}, opts = %{operation_id: _}) do
+    update_schema_and_do_call(conn, opts)
+  end
+
+  @impl Plug
+  def call(conn = %{params: %{"alert" => _}}, opts = %{operation_id: _}) do
+    update_schema_and_do_call(conn, opts)
+  end
+
+  @impl Plug
+  def call(conn = %{params: %{"data" => _}}, opts = %{operation_id: _}) do
```

Comment:
Because the `operation_id` option is introduced only after the first call of `call/2` function (the clause starting from `call(conn = %{private: %{}...})`, where it is read from `conn.private.open_api_spex` field content, so the function calls flow remains correct. The content of this file follows the original plug implementation.